polars/polars-time/src/chunkedarray/utf8/mod.rs

```diff
@@ -275,29 +277,56 @@ pub trait Utf8Methods: AsUtf8 {
                         if s.is_empty() {
                             return None;
                         }
-                        match NaiveDateTime::parse_from_str(s, fmt).map(func) {
```

Comment:
the change here looks more complicated than it is: currently, the code just uses `NaiveDateTime::parse_from_str`. With this PR, it conditionally uses either `NaiveDateTime::parse_from_str` or `DateTime::parse_from_str`, depending on `tz_aware`
Easier to review with "ignore whitespace"